scrunch/datasets.py

```diff
@@ -706,24 +731,34 @@ def move(self, path, position=-1, before=None, after=None):
 
 class Order(object):
 
-    def __init__(self, ds):
-        self.ds = ds
+    def __init__(self, resource):
+        self.resource = resource
         self._hier = None
         self._vars = None
+        self._datasets = None
         self._graph = None
         self._sync = True
         self._revision = None
 
     def _load_hier(self):
-        self._hier = self.ds.resource.session.get(
-            self.ds.resource.variables.orders.hier
-        ).payload
+        if hasattr(self.resource, 'variables'):
```

Comment:
I don't think this switch will scale well to reuse the Order class.

All shoji orders are associated to a Shoji catalog. So instead of sending the dataset or project resource you should pass in the order resource and the catalog you are ordering, in the case of a dataset you pass the variables catalog. In the case of the project you pass the datasets catalog. That way you always have `self._catalog` that should match the items in the order regardless if they are variables, datasets or something else.